fv3core/utils/grid.py

```diff
@@ -305,14 +305,14 @@ def y3d_compute_domain_x_dict(self):
         }
         return {**self.default_domain_dict(), **horizontal_dict}
 
-    def domain_shape_standard(self):
-        return (self.nid, self.njd, self.npz)
+    def domain_shape_standard(self, add: Tuple[int, int, int] = (0, 0, 0)):
+        return (self.nid + add[0], self.njd + add[1], self.npz + add[2])
 
     def domain_shape_buffer_k(self):
         return (self.nid, self.njd, self.npz + 1)
 
-    def domain_shape_compute(self):
-        return (self.nic, self.njc, self.npz)
+    def domain_shape_compute(self, add: Tuple[int, int, int] = (0, 0, 0)):
+        return (self.nic + add[0], self.njc + add[1], self.npz + add[2])
 
     def domain_shape_compute_buffer_2d(self, add: Tuple[int, int, int] = (1, 1, 0)):
```

Comment:
These two functions are now identical except for the default add value. We should replace all instances of domain_shape_buffer_2d with domain_shape_compute with the correct add if you are going to make this change.

Reply:
Done.

We should consider doing the same for the various other grid functions in a later PR, many of them can be written as a `domain_shape_compute` call (e.g. `domain_shape_compute_x` is `domain_shape_compute(add=(1, 0, 0))`.